core/src/miner/miner.rs

```diff
@@ -132,6 +132,26 @@ pub struct Miner {
     immune_users: RwLock<HashSet<Address>>,
 }
 
+struct NextAllowedReseal {
+    instant: Mutex<Instant>,
```

Comment:
If' it's okay to change Mutex into RwLock, I think we can reuse NextMandatoryReseal for it.

Reply:
It seems that we could introduce a generic structure.
However, I don't have any good name for it.

For the NextAllowedReseal and NextMandatoryReseal, I'll make `type NextMandatoryReseal = NextAllowedReseal`.